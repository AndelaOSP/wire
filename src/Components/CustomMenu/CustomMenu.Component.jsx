import React from 'react';
import PropTypes from 'prop-types';

import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';

// styles
import './CustomMenu.scss';

/**
 * @class CustomMenu
 */
class CustomMenu extends React.Component {
  constructor() {
    super();

    this.state = {
      value: 'All Countries',
    };
  }

  /**
   * Method to handle menu item selection
   */
  handleChange = (event) => {
    const { value } = event.target;
    this.props.changeCountryFilter(value);
    this.setState({ value });
  };
  

  render() {
    const styles = {
      fontSize: '0.75vw',
      color: '#000',
      backgroundColor: '#ffffff',
      width: '9.7vw',
      height: '5vh',
      textAlign: 'center',

    };

    return (
      <Select
      value={this.state.value}
      onChange={this.handleChange}
      className="custom-menu"
      style={styles}
    >
        <MenuItem value="All Countries">All Countries</MenuItem>
        <MenuItem value="Kenya">Kenya</MenuItem>
        <MenuItem value="Nigeria">Nigeria</MenuItem>
        <MenuItem value="Uganda">Uganda</MenuItem>
        <MenuItem value="USA">USA</MenuItem>
      </Select>

    ); 
  }
}

CustomMenu.propTypes = {
  changeCountryFilter: PropTypes.func,
};

CustomMenu.defaultProps = {
  changeCountryFilter: () => {},
};

export default CustomMenu;