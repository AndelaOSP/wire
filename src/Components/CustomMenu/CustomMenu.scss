.custom-menu {
  margin: 0 0.5vw;
  min-height: 2.5rem !important;
  border-radius: 4px;
  box-shadow: 0 3px 6px 0 rgba(0, 0, 0, 0.16);
  
}

.custom-menu:before,.custom-menu:after{
  border-bottom: none !important; 
}

.MuiSelect-icon-101{
  color: #000 !important;
  right: 40px !important;
}

.MuiSelect-selectMenu-99:focus{
  background-color: #fff !important;
}
