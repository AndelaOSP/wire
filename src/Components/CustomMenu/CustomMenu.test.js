import React from 'react';
import shallowToJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';
import Select from '@material-ui/core/Select';
import CustomMenu from './CustomMenu.Component';

describe('CustomMenu component', () => {
  it('Should have all the CustomMenu content', () => {
    const customMenu = shallow(<CustomMenu />);
    const tree = shallowToJSON(customMenu);
    expect(tree).toMatchSnapshot();
  });

  it('Should change state when filter value changes', () => {
    const changeCountryFilterMock = jest.fn();
    const event = { target: { value: '' } };
    const menu = shallow(<CustomMenu changeCountryFilter={changeCountryFilterMock} />);
    const select = menu.find(Select);
    select.props().onChange(event);
    expect(changeCountryFilterMock.mock.calls.length).toBe(1);
  });
});
