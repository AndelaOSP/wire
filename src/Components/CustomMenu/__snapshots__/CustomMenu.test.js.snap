// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomMenu component Should have all the CustomMenu content 1`] = `
<WithStyles(WithFormControlContext(Select))
  className="custom-menu"
  onChange={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "color": "#000",
      "fontSize": "0.75vw",
      "height": "5vh",
      "textAlign": "center",
      "width": "9.7vw",
    }
  }
  value="All Countries"
>
  <WithStyles(MenuItem)
    value="All Countries"
  >
    All Countries
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    value="Kenya"
  >
    Kenya
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    value="Nigeria"
  >
    Nigeria
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    value="Uganda"
  >
    Uganda
  </WithStyles(MenuItem)>
  <WithStyles(MenuItem)
    value="USA"
  >
    USA
  </WithStyles(MenuItem)>
</WithStyles(WithFormControlContext(Select))>
`;
